### PR TITLE
Add serialisation info for datasets

### PR DESCRIPTION
**Pull request by AtreyeeS**
This PR adds a section for serialisation in the datasets user guide as discussed in one of the previous dev calls

### CODE REVIEW

File: docs/userguide/dl4.rst [hunk 1]
```diff
@@ -218,6 +218,28 @@ even fit convergence might be an issue for a large number of datasets.
 To strike a balance, what might be a practical solution for analysis of many runs is to
 stack runs taken under similar conditions and then do a joint fit on the stacked datasets.
 
+Serialisation of datasets
+-------------------------
+
+`~gammapy.datasets.MapDataset` and `~gammapy.datasets.MapDatasetOnOff` are serialised according to `GADF Sky Maps <https://gamma-astro-data-formats.readthedocs.io/en/v0.2/skymaps/index.html>`__.
```
Comment (registerrier) [inline docs/userguide/dl4.rst:224]: Maybe: The various `Maps` contained `~gammapy.datasets.MapDataset` and `~gammapy.datasets.MapDatasetOnOff` are serialized according

File: docs/userguide/dl4.rst [hunk 2]
```diff
@@ -218,6 +218,28 @@ even fit convergence might be an issue for a large number of datasets.
...
+A hdulist is cretaed with the different attributes, and each of these are written with the data
```
Comment (registerrier) [inline docs/userguide/dl4.rst:225]: created

**Review by registerrier — APPROVED** [review]
Thanks @AtreyeeS . I have left two minor comments inline

File: docs/userguide/dl4.rst (hunk 2 above)
Comment (adonath) [inline docs/userguide/dl4.rst:225]:
```suggestion
A hdulist is created with the different attributes, and each of these are written with the data
```

File: docs/userguide/dl4.rst (hunk 1 above)
Comment (adonath) [inline docs/userguide/dl4.rst:224]:
```suggestion
The various `Map` objects contained in `~gammapy.datasets.MapDataset` and `~gammapy.datasets.MapDatasetOnOff` are serialised according to `GADF Sky Maps <https://gamma-astro-data-formats.readthedocs.io/en/v0.2/skymaps/index.html>`__.
```

**Review by adonath — APPROVED** [review]
Thanks @AtreyeeS!

**Comment by adonath** [comment]
I addressed @registerrier comments, I'll wait for the CI to pass and then merge.

**Pull request merged**